pandas/core/groupby/groupby.py

```diff
@@ -353,13 +355,16 @@ def _group_selection_context(groupby):
 ]
 
 
-class _GroupBy(PandasObject, SelectionMixin):
+_GroupByT = TypeVar("_GroupByT", bound="NDFrame")
+
+
+class _GroupBy(Generic[_GroupByT], PandasObject, SelectionMixin):
```

Comment:
can use FrameOrSeries, see other comment.

Reply:
Sorry - I don't quite follow with the other comment. If one uses 

    class _GroupBy(Generic[FrameOrSeries], ...) 

here instead of a new type _GroupByT, then when calling class SeriesGroupBy[Series] you would have FrameOrSeries being Series. Now some methods of SeriesGroupBy use FrameOrSeries as the return type because they may either return a DataFrame or Series, but now the return type is simply Series which is incorrect.

Reply:
> Now some methods of SeriesGroupBy use FrameOrSeries as the return type because they may either return a DataFrame or Series,

If the return type of a Groupby method cannot be determined by the class type, then FrameOrSeries is incorrect and needs to be replaced with FrameOrSeriesUnion.

Reply:
i find the name _GroupByT confusing since the typevar is bound by NDFrame

Reply:
> Now some methods of SeriesGroupBy use FrameOrSeries as the return type because they may either return a DataFrame or Series

like ohlc?

Reply:
see commit linked in https://github.com/pandas-dev/pandas/pull/32302#issuecomment-594541073 for how to handle ohlc

Reply:
Ah, I missed that there was a commit in the comment you linked to. Perfectly clear now, and changes made. One thing I did differently is named it "OutputFrameOrSeries" instead of "FrameOrSeries2". Thanks!